datafusion/common/src/file_options/csv_writer.rs

```diff
@@ -41,6 +43,20 @@ impl CsvWriterOptions {
         Self {
             writer_options,
             compression,
+            compression_level: None,
+        }
+    }
+
+    /// Create a new `CsvWriterOptions` with the specified compression level.
+    pub fn new_with_level(
+        writer_options: WriterBuilder,
+        compression: CompressionTypeVariant,
+        compression_level: Option<u32>,
```

Comment:
Is it better to just have `compression_level: u32` and direct users to use `new` if they want default (`None`) compression level? Thoughts? 🤔